### ISSUE TITLE
Compatibility with Vite 6

### ISSUE THREAD

**Issue by marcphilipp** [richardtallent/vite-plugin-singlefile#104]
I'm getting the following error message when running `npm install` after updating the `vite` dependency to 6.0.0:

```
npm error While resolving: vite-plugin-singlefile@2.0.3
npm error Found: vite@6.0.0
npm error node_modules/vite
npm error   dev vite@"6.0.0" from the root project
npm error   peer vite@"^5.0.0 || ^6.0.0" from @vitejs/plugin-vue@5.2.1
npm error   node_modules/@vitejs/plugin-vue
npm error     dev @vitejs/plugin-vue@"5.2.1" from the root project
npm error
npm error Could not resolve dependency:
npm error peer vite@"^5.4.10" from vite-plugin-singlefile@2.0.3
npm error node_modules/vite-plugin-singlefile
npm error   dev vite-plugin-singlefile@"2.0.3" from the root project
npm error
npm error Conflicting peer dependency: vite@5.4.11
npm error node_modules/vite
npm error   peer vite@"^5.4.10" from vite-plugin-singlefile@2.0.3
npm error   node_modules/vite-plugin-singlefile
npm error     dev vite-plugin-singlefile@"2.0.3" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```

Does the plugin work with 6.0.0? Do you have plans to update the peer dependency?

**Comment by richardtallent** [comment]
Thanks! I literally just read the Vite 6 announcement. Currently visiting family for Thanksgiving and likely won't have time to play with compatibility fixes and/or the new API this week.

Reproducible issue repositories are appreciated -- can't spend time copying and pasting code from Issues.

**Comment by marcphilipp** [comment]
Here's a reproducer: https://github.com/marcphilipp/vite-6-plugin-singlefile-reproducer

**Comment by richardtallent** [comment]
The PR needed some tweaking. I've released 2.1 here, but npm is under maintenance at the moment, so I haven't been able to publish it. I'll try again in a few hours.

**Comment by richardtallent** [comment]
Published 2.1.0

**Comment by marcphilipp** [comment]
Awesome, thank you! :tada:
